lib/galaxy/jobs/runners/local.py

```diff
@@ -116,11 +118,9 @@ def queue_job(self, job_wrapper):
                 self._handle_container(job_wrapper, proc)
 
                 terminated = self.__poll_if_needed(proc, job_wrapper, job_id)
+                proc.wait()  # reap
```

Comment:
Actually no, before `proc.wait()` was called inside `self._terminate(proc)` by `self.__poll_if_needed()` before returning `True`.
Now `kill_pg()` receives `pgid` instead of `proc`, so reap needs to be done here.